site/Cargo.toml

```diff
@@ -57,3 +57,6 @@ jemalloc-ctl = "0.5"
 serde = { workspace = true, features = ["derive"] }
 serde_json = { workspace = true }
 toml = "0.7"
+
+[dev-dependencies]
+insta = "1.40.0"
```

Comment:
do we want/need insta? Managing the snapshots files can be cumbersome, and it's yet another tool to use and learn. It's fine if we want to make good use of it in the future, and not a fancy `assert_eq!(format!("{obj:?}"), "Struct { field: 'bla'} ")`.

Reply:
Since I'm refactoring this, I want to do it properly. Note that I'm not using snapshot files, I'm using the inline snapshots, which makes this much easier to grok, I think.

I'm already anticipating future changes (like your PR to add the `backends`) parameter. It's no fun updating tens of tests anytime you change the structure of the thing that you parse. Snapshot testing makes this much easier.

I will document in readme what to do to update the snapshots though.

Reply:
Insta adds snapshot files locally when a test fails, the pending snap file. I saw that when adding the test I shared. It didn’t remove it when the test passed.

Reply:
Ah, these things. Right, I guess we can add them to `.gitignore`, but they should be removed after you use `cargo insta review` I think (the snapshots shouldn't really be modified manually).

Reply:
I didn't have `insta` installed to do a `cargo insta review` anyways. We're testing a struct with 3 fields, we don't really need all this test infra imho.

Reply:
There are already 23 tests in the file, I ain't gonna rewrite all the asserts to make sure we parse exactly what we think we do, every time we change the parsing code :laughing: `insta` was already super helpful for me when developing this PR.